contracts/external/proxies/SignedOperationProxy.sol

```diff
@@ -0,0 +1,550 @@
+    }
+
+     /**
+     * The owner can start back up the exchange.
+     */
+    function startUp()
+        external
+        onlyOwner
+    {
+        g_isOperational = true;
+        emit ContractStatusSet(true);
+    }
+
+    // ============ Public Functions ============
+
+    /**
+     * Allows a signer to permanently cancel an operation on-chain.
+     *
+     * @param  operationHash  The EIP712 hash of the Operation message to cancel.
+     */
+    function cancel(
+        bytes32 operationHash
+    )
+        external
+    {
+        g_invalidated[msg.sender][operationHash] = true;
+        emit LogOperationCanceled(operationHash, msg.sender);
+    }
+
+    /**
+     * Submits an operation to SoloMargin. Actions for accounts that the msg.sender does not control
+     * must be authorized by a signed message. Each authorization can apply to multiple actions at
+     * once which must occur in order next to each other. An empty authorization must be supplied
+     * explicitly for each group of actions that do not require a signed message.
+     *
+     * @param  accounts  The accounts to forward to SoloMargin.operate()
+     * @param  actions   The actions to forward to SoloMargin.operate()
+     * @param  auths     The signed authorizations for each group of actions
+     *                   (or unsigned if msg.sender is already authorized)
+     */
+    function operate(
+        Account.Info[] memory accounts,
+        Actions.ActionArgs[] memory actions,
+        Authorization[] memory auths
+    )
+        public
+    {
+        Require.that(
+            g_isOperational,
+            FILE,
+            "Contract is not operational"
+        );
+
+        // cache the index of the first action for this auth
+        uint256 authStartIndex = 0;
+
+        // loop over all auths
+        for (uint256 p = 0; p < auths.length; p++) {
+            Authorization memory auth = auths[p];
+
+            // require that the message is not expired
+            Require.that(
+                auth.expiration == 0 || auth.expiration >= block.timestamp,
+                FILE,
+                "Signed operation is expired",
+                p
+            );
+
+            // require that the sender is the sender
+            Require.that(
+                auth.sender == address(0) || auth.sender == msg.sender,
+                FILE,
+                "Operation sender mismatch",
+                p
+            );
+
+            // get the signer of the auth (msg.sender if no auth)
+            address signer = getSigner(
+                accounts,
+                actions,
+                auth,
+                authStartIndex
+            );
+
+            // cache the index of the first action after this auth
+            uint256 authEndIndex = authStartIndex.add(auth.numActions);
+
+            // loop over all actions for which this auth applies
+            for (uint256 a = authStartIndex; a < authEndIndex; a++) {
```

Comment:
if `authEndIndex` (`a`) is out of bounds for `actions` solidity will throw an error right?